hub/hub.ml

```diff
@@ -0,0 +1,113 @@
+(** See https://hub.github.com/hub.1.html *)
+open Cmdliner
+open Printf
+
+type token = {
+    token: Github.Token.t;
+    user: string
+  }
+
+exception Config of string
+
+let config_from_file () =
+  let home = Unix.getenv "HOME" in
+  let fpath = home ^ "/.config/hub" in
+  let result = Yaml_unix.of_file_exn Fpath.(v fpath)  in
```

Comment:
Probably worth catching the exception here (or using `Yaml_unix.of_file`) to give a less inscrutable error in case the config file isnt there or readable.

Reply:
I'll add a catch with some reasonable error message, thanks for the suggestion.

The `match result` code is pretty unsatisfying, I really want to reach for a more generic solution to de-structuring the YAML type like `Yojson.Basic.Util` or lenses 😲